src/gcp/iam.ts

```diff
@@ -222,3 +219,33 @@ export async function testIamPermissions(
     `projects/${projectId}`,
   );
 }
+
+/**
+ * Generate the updated list of IAM bindings given the existing and new IAM bindings
+ * @param existingBindings The existing IAM bindings
+ * @param newBindings The new IAM bindings to update the existing one with
+ */
+export function generateUpdatedIamBindings(
+  existingBindings: Binding[],
+  newBindings: Binding[],
+): Binding[] {
+  const updatedBindings: Binding[] = existingBindings ? [...existingBindings] : [];
+
+  for (const newBinding of newBindings) {
+    const existingRoleIndex = updatedBindings.findIndex((b) => b.role === newBinding.role);
+
+    if (existingRoleIndex >= 0) {
```

Comment:
Can you double check this is necessary? I _think_ IAM dedupes/coalesces roles but it's been a while since I've played with the API.

Also, tests? :)

Reply:
Ahh you're right this is completely unnecessary, let's let the API handle it. Removed the method and the secrets.spec.ts has a test to ensure this